ROADMAP.md

```diff
@@ -24,10 +27,102 @@ Add @salt{1%tsp} -- this is the same
 Add @salt{1|2|4%tsp} -- as this
 ```
 
-Alternatively, you can define the same quantity for all serving sizes with `!`.
+Mentioned serving options will be available in UI. But it's possible to scale to a number of servings which is not defined here. The parser will inter- or extrapolate if necessary.
+
+Default value for servings will be the first one. It's not required to go always up or down.
+
+If servings not specified, it considered to be 1.
+
+#### TODO
+
+* How to scale cookware, timer?
+* How it works with include
+
+### Optional ingredients
+
+Use `?` at the end of an ingredient name to indicate that an ingredient is not required:
+
+```
+Add some @?rum and @?egg yolk{1}, if desired.
+```
+
+#### TODO
+
+* Can ingredients contain a question mark? Will it mess it up?
+* Other syntax idea:
+
+```
+Add some @rum? and @egg yolk?{1}, if desired.
+```
+
+
+### Variants
+
+Separate ingredients with '|' to define alternatives.
+
+```
+Rinse the @bream{450%g} | @cod{450%g} | @seabass{300%g} | @catfish{450%g}, remove any scales that may remain, as well as fins.
```

Comment:
Different ingredients may introduce different timer for cooking, how do we resolve this?